### DISCUSSION TITLE
How to work with point clouds?

### DISCUSSION THREAD

**Discussion by ZelasYou** [mozman/ezdxf#1118] [Q&A] [answered]
**Describe the bug**
Color point cloud conversion to. dxf cannot be displayed in AutoCAD

**To Reproduce**
```python
    # 创建一个新的 DWG 文档
    doc = ezdxf.new(dxfversion='R2013')
    msp = doc.modelspace()
    # doc.header.custom_vars.replace('$DWGCODEPAGE','ANSI_936')
    # print(doc.header.hdrvars.keys())
    # doc.header.hdrvars['$DWGCODEPAGE'] = 'ANSI_936'  # 设置中文字符
    # 遍历 LAS 文件中的每个点
    # 设置新的边界点位置
    extmin = (np.min(las.x), np.min(las.y), np.min(las.z))
    extmax = (np.max(las.x), np.max(las.y), np.max(las.z))
    limmax = (np.max(las.x) + 1, np.max(las.y) + 1)
    limmin = (np.min(las.x) - 1, np.min(las.y) - 1)
    # 将边界点位置添加到 HEADER 段中
    msp.reset_extents(extmin, extmax)
    msp.reset_limits(limmin, limmax)
    for x, y, z, r, g, b in tqdm(zip(las.x, las.y, las.z, las.red, las.green, las.blue), total=len(las.xyz)):
        # 将颜色转换为 AutoCAD 颜色
        color = colors.rgb2int((normalize_color(r), normalize_color(g), normalize_color(b)))
        # print('当前颜色为',color)
        # 添加点到 DXF 文件
        msp.add_point((x, y, z))
        # 将点添加到 DXF 文件中
        dxf_point = msp.add_point((x, y, z))
        dxf_point.dxf.owner = '17'  # 示例拥有对象句柄
        # 设置颜色（使用颜色值的直接设置）
        # 设置其他属性
        dxf_point.dxf.linetype = 'Continuous'
        dxf_point.dxf.color = 256  
        dxf_point.dxf.true_color = color  
        dxf_point.dxf.lineweight = 13  # 线宽
    doc.saveas('las2dxf.dxf')
```

**Expected behavior**
Usually, if the output is successful, a colored point cloud can be displayed in CAD
![1e71ba3666de60d21bdf67fe81492b1](https://github.com/mozman/ezdxf/assets/89770548/8f02dba8-b67d-49cb-a31f-e95ab7f2625b)
But when I open it, it doesn't show anything, but it can be opened and displayed normally in software such as ArcGIS and Global Mapper. I want to know what the problem is, thank you.

**Comment by mozman** [comment]
Hi!

1. Don't set the owner tag by yourself if don't know what you do: `dxf_point.dxf.owner = '17'`
2. ~Use the `ezdxf.addons.odafc` add-on to export DWG files~
3. I don't think a DXF file with millions of `POINT` entities will work as expected
4. `ezdxf` has no features to work with point clouds

**Reply by ZelasYou** [reply]
I'm very sorry, author. I didn't organize my language well at the beginning. Additionally, I have provided a more detailed explanation in your email. The ability to display colored point clouds through other software indicates that ezdxf is effective.The. dxf fields that display color point clouds normally are as follows:
POINT
  5
692C98
330
11FB49E
100
AcDbEntity
  8
0
  6
Continuous
 62
   250
420
  1579032
370
    13
100
AcDbPoint
 10
507386.1800000001
 20
4327870.76
 30
-5.13
  0
And after converting my fields through ezdxf, they only show a different order, so I suspect that there may be a problem with the header or ending files. I changed a few properties but did not change, and I cannot find the reason

**Reply by ZelasYou** [reply]
The correct display and incorrect display of. dxf have been shared with you through the online storage link:
https://drive.google.com/file/d/1J2MfPNPMWLlsQpzdMw3EX1ksv-pO6jLv/view?usp=sharing, https://drive.google.com/file/d/1KaQyAiOpR71tGGJsfD8og7eF6TGDO7yR/view?usp=sharing

**Reply by mozman** [reply]
This is a minimal version of your code and that is all `ezdxf` can do:

```python
import ezdxf
from tqdm import tqdm
import laspy as lp


def normalize_color(value):
    return int((value / 65535) * 255)

if __name__ == "__main__":
    las = lp.read('point_cloud.las')
    doc = ezdxf.new(dxfversion='R2013')
    msp = doc.modelspace()
    for x, y, z, r, g, b in tqdm(zip(las.x, las.y, las.z, las.red, las.green, las.blue), total=len(las.xyz)):
        dxf_point = msp.add_point((x, y, z))
        dxf_point.rgb = (normalize_color(r), normalize_color(g), normalize_color(b))
    doc.saveas('las2dxf.dxf')
```

If this doesn't work, you need another tool!

**Comment by mozman** [comment, marked as answer]
SOLUTION:

Enter `ZOOM E` in the command line of your CAD application

![image](https://github.com/mozman/ezdxf/assets/4230966/88beec90-1dce-4960-8b2f-bc8c4f3a480f)

**Reply by ZelasYou** [reply]
Thank you very much. You have successfully solved the problem
